minari/dataset/minari_storage.py

```diff
@@ -87,17 +88,53 @@ def apply(
 
         return out
 
+    def _h5_group_to_dict_recursive(self, hdf_ref, timestep):
+
+        if not isinstance(hdf_ref, h5py.Group):
+            return hdf_ref[timestep]
+
+        if "tuple" in hdf_ref.attrs.keys():
+            result = []
+            for i in range(len(hdf_ref.keys())):
+                result.append(
+                    self._h5_group_to_dict_recursive(hdf_ref[f"_index_{i}"], timestep)
+                )
+            return tuple(result)
+        else:
+            result = {}
+            for key in hdf_ref:
+                result[key] = self._h5_group_to_dict_recursive(hdf_ref[key], timestep)
+            return result
```

Comment:
the tuple and dict branches can be merged and call tuple on values if the space is a tuple

Reply:
The reasoning behind the explicit iteration is to avoid having to perform a sort by key which I think might be necessary if grabbing the values to construct the tuple.

Reply:
I see; we can use `OrderedDict` for tuples, but it is not needed for now (if it happens that we have many different codes that can be merged we should switch to it)

Reply:
I see; we can use OrderedDict for tuples, but it is not needed for now (if it happens that we have many different codes that can be merged we should switch to it)